sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -232,16 +232,29 @@ Command *Scheduler::GraphBuilder::insertMemoryMove(MemObjRecord *Record,
 
   AllocaCommandBase *AllocaCmdSrc =
       findAllocaForReq(Record, Req, Record->MCurContext);
+  if (IsSuitableSubReq(Req) && !AllocaCmdSrc) {
+    auto IsSuitableAlloca = [Record, Req](AllocaCommandBase *AllocaCmd) {
+      bool Res = sameCtx(AllocaCmd->getQueue()->get_context_impl(),
+                         Record->MCurContext) &&
+                 AllocaCmd->getRequirement()->MSYCLMemObj == Req->MSYCLMemObj;
+      return Res;
+    };
+    const auto It =
+        std::find_if(Record->MAllocaCommands.begin(),
+                     Record->MAllocaCommands.end(), IsSuitableAlloca);
+    AllocaCmdSrc = (Record->MAllocaCommands.end() != It) ? *It : nullptr;
+  }
   if (!AllocaCmdSrc)
     throw runtime_error("Cannot find buffer allocation");
   // Get parent allocation of sub buffer to perform full copy of whole buffer
   if (IsSuitableSubReq(Req)) {
     if (AllocaCmdSrc->getType() == Command::CommandType::ALLOCA_SUB_BUF)
       AllocaCmdSrc =
           static_cast<AllocaSubBufCommand *>(AllocaCmdSrc)->getParentAlloca();
+    else if (AllocaCmdSrc->getSYCLMemObj() == Req->MSYCLMemObj)
+      AllocaCmdSrc = static_cast<AllocaCommand *>(AllocaCmdSrc);
     else
```

Comment:
What this code does? To me it looks like a NOP.

Reply:
Refactored. There is a need to ensure, that found alloca command relates to a current requirement.